### ISSUE TITLE
[Sort] Animations not playing correctly

### ISSUE THREAD

**Issue by nickwinger** [angular/components#9939]
#### Bug:
When setting the initial sort with the Method sort on MatSort, the arrow is not shown, it's opactiy
is 0

#### What is the expected behavior?
The arrow should be visible when setting the initial sorted column with the method sort on MatSort

#### What is the current behavior?
The opacity of the arrow remains 0 although the column is set to sorted

#### What are the steps to reproduce?
Providing a StackBlitz reproduction is the *best* way to share your issue. <br/>
StackBlitz starter: https://goo.gl/wwnhMV<br/>
Create a Component with a mat-table and mat-sort (with changeDetection PUSH)
Get the MatSort via ViewChild and set the default sorted column in ngOnInit with the method sort

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Angular 5.2.4
Angular Material 5.2.0
Windows 10
Typescript 2.6.2
Chrome Browser 64

#### Is there anything else we should know?
It has already worked... (some Version ago)

**Comment by Manduro** [comment]
@nickwinger Please provide a reproduction.

> **What are the steps to reproduce?**
> 
> Providing a StackBlitz reproduction is the best way to share your issue. 
> 
> StackBlitz starter: https://goo.gl/wwnhMV

I've tried to reproduce it following your steps, but it seems to be working correctly:
https://stackblitz.com/edit/angular-material2-issue-stefcz?file=app%2Fapp.component.ts

**Comment by shahmirn** [comment]
@Manduro I was able to reproduce it by updating material and wrapping the code in a timer(0)

https://angular-material2-issue-mc4cve.stackblitz.io

Seems to be a regression from @angular/material 5.1.0

**Comment by AmandaFoster** [comment]
Since they are probably related, this same issue occurs when setting the sort via the `matSortDirection` and `matSortActive` inputs.

https://stackblitz.com/edit/angular-w13pnn?file=app%2Fsort-overview-example.ts

**Comment by danielhdz56** [comment]
A quick fix would be to do something like:
```
.mat-sort-header-sorted {
  .mat-sort-header-arrow {
    opacity: 1 !important;
    transform: translateY(0px) !important;
  }
}

```

**Comment by josephperrott** [comment]
Closing as this was fixed by: #10593
